src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageCodeFix.cs

```diff
@@ -29,21 +38,45 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
             var diagnostic = context.Diagnostics.First();
             var invocationNode = root.FindNode(diagnostic.Location.SourceSpan) as InvocationExpressionSyntax;
-            var invocationIdentifier = diagnostic.Properties[AnalyzerPropertyKeys.ModelName];
 
             if (invocationNode == null)
                 return;
 
+            var invocationIdentifier = diagnostic.Properties[AnalyzerPropertyKeys.ModelName];
+            var isGenericMethod = diagnostic.Properties[AnalyzerPropertyKeys.IsGenericMethod] == true.ToString();
+
             // First, replace the original method invocation name to "That".
+            var descendantNodes = invocationNode.DescendantNodes(_ => true).ToArray();
+
+            SyntaxNode methodNode;
+            TypeArgumentListSyntax? typeArguments;
+            if (isGenericMethod)
+            {
+                methodNode = descendantNodes.Where(_ => _.IsKind(SyntaxKind.GenericName) &&
+                                                        ((GenericNameSyntax)_).Identifier.Text == invocationIdentifier)
+                                            .Single();
+                typeArguments = descendantNodes.Where(_ => _.IsKind(SyntaxKind.TypeArgumentList))
+                                               .Cast<TypeArgumentListSyntax>()
+                                               .Single();
```

Comment:
This will not work, if the generic has nested type paramters, e.g. `Assert.IsInstanceOf<List<int>>(x);`. Then it fails with 
```
System.InvalidOperationException : Sequence contains more than one element
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at async NUnit.Analyzers.ClassicModelAssertUsage.ClassicModelAssertUsageCodeFix.RegisterCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```
We should take the outermost one (I guess this is the first, but I've not checked).

Reply:
@mikkelbu Thanks for that. I have added test for nested generics and fixed the code. You were right, it was the First.
and fixed the indentations.
Also added a commit to fix the expect/actual order in the .md files for Greater/Less. Next time I will proof read my changes better before raising a pull request.